scipy/stats/_stats_py.py

```diff
@@ -496,7 +496,21 @@ def pmean(a, p, *, axis=0, dtype=None, weights=None):
 ModeResult = namedtuple('ModeResult', ('mode', 'count'))
 
 
-def mode(a, axis=0, nan_policy='propagate', keepdims=None):
+def _mode_result(mode, count):
+    # When a slice is empty, `_axis_nan_policy` automatically produces
+    # NaN for `mode` and `count`. This is a reasonable convention for `mode`,
+    # but `count` should not be NaN; it should be zero.
+    i = np.isnan(count)
+    if i.shape == ():
+        count = count.dtype(0) if i else count
+    else:
+        count[i] = 0
+    return ModeResult(mode, count * (~np.isnan(count)))
```

Comment:
Oops, this can be simplified. 
```suggestion
    return ModeResult(mode, count)
```
I was playing around with removing all of the code above and just doing `count * (~np.isnan(count))`, but `np.nan * 0` is `np.nan`, not `0`.